linkerd/http-route/src/http/filter/modify_request_header.rs

```diff
@@ -0,0 +1,24 @@
+use http::header::{HeaderMap, HeaderName, HeaderValue};
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct ModifyRequestHeader {
+    pub add: Vec<(HeaderName, HeaderValue)>,
+    pub set: Vec<(HeaderName, HeaderValue)>,
+    pub remove: Vec<HeaderName>,
```

Comment:
it makes sense for `add` to be a `Vec`, since we might want to add multiple values for the same header name, but i wonder if `set` (and `remove`?) ought to be a `HashMap` or something? allowing multiple entries in `set` for the same header name feels kinda sketchy, since then we would end up with a final header value that depends on the order of the entries in the `Vec`...and for removing headers, removing the same header name multiple times is just an unnecessary hashmap lookup...